src/analysis/buildTypeWeights.ts

```diff
@@ -19,168 +21,199 @@ import {
-                        (cur) => cur.name.value === arg.name
-                    );
-
-                    if (limitArg) {
-                        const node: ValueNode = limitArg.value;
-
-                        if (Kind.INT === node.kind) {
-                            const multiplier = Number(node.value || arg.defaultValue);
-                            const weight = isCompositeType(listType)
-                                ? result[listType.name.toLowerCase()].weight
-                                : typeWeights.scalar || DEFAULT_SCALAR_WEIGHT; // Note this includes enums
-
-                            return weight * multiplier;
-                        }
-
-                        if (Kind.VARIABLE === node.kind) {
-                            // TODO: Get variable value and return
-                            // const multiplier: number =
-                            // return result[listType.name.toLowerCase()].weight * multiplier;
-                            throw new Error(
-                                'ERROR: buildTypeWeights Variable arge values not supported;'
-                            );
-                        }
-                    }
-
-                    // FIXME: The list is unbounded. Return the object weight for
-                    throw new Error(
-                        `ERROR: buildTypeWeights: Unbouned list complexity not supported. Query results should be limited with ${KEYWORDS}`
-                    );
-                };
-            }
-        });
+    result.query = parseObjectFields(queryType, typeWeightObject, typeWeights);
 
-        // if the field is a scalar or an enum set weight accordingly. It is not a list in this case
-        if (isScalarType(resolveType) || isEnumType(resolveType)) {
-            result.query.fields[field] = typeWeights.scalar || DEFAULT_SCALAR_WEIGHT;
-        }
-    });
     return result;
 }
 
 /**
  * Parses all types in the provided schema object excempt for Query, Mutation
  * and built in types that begin with '__' and outputs a new TypeWeightObject
  * @param schema
- * @param typeWeightObject
  * @param typeWeights
  * @returns
  */
-function parseTypes(
-    schema: GraphQLSchema,
-    typeWeightObject: TypeWeightObject,
-    typeWeights: TypeWeightConfig
-): TypeWeightObject {
+function parseTypes(schema: GraphQLSchema, typeWeights: TypeWeightConfig): TypeWeightObject {
     const typeMap: ObjMap<GraphQLNamedType> = schema.getTypeMap();
 
-    const result: TypeWeightObject = { ...typeWeightObject };
+    const result: TypeWeightObject = {};
 
     // Handle Object, Interface, Enum and Union types
     Object.keys(typeMap).forEach((type) => {
         const typeName: string = type.toLowerCase();
-
         const currentType: GraphQLNamedType = typeMap[type];
+
         // Get all types that aren't Query or Mutation or a built in type that starts with '__'
         if (type !== 'Query' && type !== 'Mutation' && !type.startsWith('__')) {
```

Comment:
We can probably remove the check for `Query` here and then remove the `parseQueryType` function entirely.

Reply:
I've done this. Good call. It was very easy and will handle mutations automatically. So we are now parsing types and parsing the fields on the types which a good direction to go in for better modularity.